client/client_test.go

```diff
@@ -0,0 +1,94 @@
+package client
+
+import (
+	"testing"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+	"github.com/heroku/silvia-runtime-university/spec"
+	"reflect"
+	"fmt"
+	"strings"
+)
+
+var testCases = []struct{
+	name string
+	inputPoints []spec.Point
+	wantFeatures []spec.Feature
+}{
+	{
+		name:         "With zero points",
+		inputPoints:  []spec.Point{},
+		wantFeatures: []spec.Feature{},
+	},
+	{
+		name:        "With one point",
+		inputPoints: []spec.Point{{ Latitude: 100001, Longitude: 100002},
+		},
+		wantFeatures: []spec.Feature{
+			{ Name: "somefeature", Location: &spec.Point{Latitude: 100001, Longitude: 100002} },
+		},
+	},
+	{
+		name:        "With two points",
+		inputPoints: []spec.Point{{ Latitude: 100001, Longitude: 100002}, {Latitude: 100003, Longitude: 100004 }},
+		wantFeatures: []spec.Feature{
+			{ Name: "somefeature", Location: &spec.Point{Latitude: 100001, Longitude: 100002} },
+			{ Name: "anotherfeature", Location: &spec.Point{Latitude: 100003, Longitude: 100004} },
+		},
+	},
+	{
+		name:        "With point unassociated to feature",
+		inputPoints: []spec.Point{{ Latitude: 100008, Longitude: 100009}},
+		wantFeatures: []spec.Feature{
+			{ Name: "somefeature", Location: &spec.Point{Latitude: 100001, Longitude: 100002} },
+		},
+	},
+}
+
+func TestGetFeatures(t *testing.T) {
+
+	for _, test := range testCases {
+		t.Run(test.name, func(t *testing.T) {
+			var routeGuide RouteGuide
+			ctx := context.Background()
+			routeGuide.client = fakeRouteGuideClient{ features: test.wantFeatures }
+
+			gotFeatures, err := routeGuide.GetFeatures(ctx, test.inputPoints)
+			if err == nil {
```

Comment:
It's common in Go to deal with the error cases first, and then look at the success case. When the error case causes the program flow to stop (`t.Fatalf`, `return err`, etc.), the else even becomes unnecessary. This style is described as [Indent error flow](https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow) and shows up all the time in Go code.